csharp/src/Ice/Proxy.cs

```diff
@@ -322,76 +325,170 @@ public static class Proxy
                 IInvocationObserver? observer = ObserverHelper.GetInvocationObserver(proxy,
                                                                                      request.Operation,
                                                                                      request.Context);
-                int retryCount = 0;
                 try
                 {
+                    IncomingResponseFrame? response = null;
+                    Exception? lastException = null;
+                    Retryable retryable = Retryable.No;
+                    TimeSpan afterDelay = TimeSpan.Zero;
+                    var excludedConnectors = new List<IConnector>();
```

Comment:
We could lazy initialize the list instead of allocating it all the time even if it's not going to b used most of the time.

Reply:
Yes that makes sense, will do